ibis/expr/operations.py

```diff
@@ -284,6 +284,11 @@ class UnaryOp(ValueOp):
     arg = Arg(rlz.any)
 
 
+class BinaryOp(ValueOp):
+    left = Arg(rlz.any)
+    right = Arg(rlz.any)
```

Comment:
There already exists a [BinaryOp](https://github.com/ibis-project/ibis/blob/master/ibis/expr/operations.py#L702). 

However I think this is the proper implementation for `BinaryOp` and the old one should be removed. If I recall correctly [Comparison](https://github.com/ibis-project/ibis/blob/master/ibis/expr/operations.py#L2048) is the only operation which depends on calling `_maybe_cast_args`. 

So if You move [BinaryOp.__init__](https://github.com/ibis-project/ibis/blob/master/ibis/expr/operations.py#L711) to [Comparison](https://github.com/ibis-project/ibis/blob/master/ibis/expr/operations.py#L2048) along with the [comments](https://github.com/ibis-project/ibis/blob/master/ibis/expr/operations.py#L705) then We could delete the [old BinaryOp](https://github.com/ibis-project/ibis/blob/master/ibis/expr/operations.py#L702) and use the one defined here instead.